### PR TITLE
fix(notice): add ability to dismiss Give notices on front-end. #2853

### LINKED ISSUE #2853: fix(notice): Dismiss button is missing in Give notices in front-end.
## Issue Overview

Notice in front-end area can't be dismissed 

`Give_Notices::print_frontend_notice( __( 'Sample notice', 'give' ), true );`

In below screenshot there is no dismiss button.
screenshot-givecurrencyswitcher com-2018 02 28-16-28-48

## Expected Behavior

- Notices in front-end should be dismiss.

## Current Behavior

- Dismiss button is not there in front-end GIve notices.

## Todos
- [ ] Tests
- [ ] Documentation

## WordPress Environment
<details>
```
### WordPress Environment ###

Home URL: http://givecurrencyswitcher.com
Site URL: http://givecurrencyswitcher.com
WP Version: 4.9.4
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
Show on Front: posts
Table Prefix Length: wp_
Table Prefix Length: 3
Table Prefix Status: Acceptable
Admin AJAX: Accessible
Registered Post Statuses: publish, future, draft, pending, private, trash, auto-draft, inherit, refunded, failed, revoked, cancelled, abandoned, processing, preapproval

### Server Environment ###

Hosting Provider: DBH: localhost, SRV: givecurrencyswitcher.com
TLS Connection: Connection uses TLS 1.2
TLS Connection: Probably Okay
Server Info: nginx/1.10.1
PHP Version: ❌ 5.3.29 - We recommend a minimum PHP version of 5.6. See: PHP Requirements in Give
PHP Post Max Size: 1,000 MB
PHP Time Limit: 900
PHP Max Input Vars: 3000
PHP Max Upload Size: 1,000 MB
cURL Version: ❌ 7.38.0, OpenSSL/1.0.2h - We recommend a minimum cURL version of 7.40.
SUHOSIN Installed: –
MySQL Version: ❌ 5.5.55 - We recommend a minimum MySQL version of 5.6. See: WordPress Requirements
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
gzip: ✔
GD Graphics Library: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Give Configuration ###

Give Version: 2.0.5
Give Cache: Enabled
Database Updates: All DB Updates Completed.
Give Cache: Enabled
Give Cache: ✔New Donation✔Donation Receipt❌New Offline Donation❌Offline Donation Instructions✔New User Registration✔User Registration Information✔Email access
Upgraded From: 2.0.2
Test Mode: Enabled
Currency Code: GBP
Currency Position: Before
Decimal Separator: .
Thousands Separator: ,
Success Page: http://givecurrencyswitcher.com/donation-confirmation/
Failure Page: http://givecurrencyswitcher.com/donation-failed/
Donation History Page: http://givecurrencyswitcher.com/donation-history/
Give Forms Slug: /donations/
Enabled Payment Gateways: Test Donation, Stripe - Credit Card, GoCardless, Authorize.net, Mollie Gateway, PayPal Payments Pro
Default Payment Gateway: Test Donation
PayPal IPN Verification: Enabled
PayPal IPN Notifications: N/A
Donor Email Access: Enabled

### Session Configuration ###

Give Use Sessions: Enabled
Session: Disabled

### Active Give Add-ons ###

Give - Authorize.net Gateway: ❌ Unlicensed – by WordImpress – 1.3.3
Give - Currency Switcher: ❌ Unlicensed – by WordImpress – 1.0.1
Give - Fee Recovery: ❌ Unlicensed – by WordImpress – 1.4
Give - Form Field Manager: ❌ Unlicensed – by WordImpress – 1.2.8
Give - GoCardless Gateway: ❌ Unlicensed – by WordImpress – 1.1.2
Give - Manual Donations: ❌ Unlicensed – by WordImpress – 1.3.1
Give - Mollie Gateway: ❌ Unlicensed – by WordImpress – 1.0
Give - PayPal Pro Gateway: ❌ Unlicensed – by WordImpress – 1.1.4
Give - PDF Receipts: ❌ Unlicensed – by WordImpress – 2.2.4
Give - Razorpay: ❌ Unlicensed – by WordImpress – 1.1.1
Give - Stripe Gateway: ❌ Unlicensed – by WordImpress – 1.5.2

### Other Active Plugins ###


### Inactive Plugins ###

Debug Bar: by wordpressdotorg – 0.9
Debug Bar Post Meta: by whyisjake – 0.5.4
Duplicate Post: by Enrico Battocchi – 3.2.1
Give - Tributes: by WordImpress – 1.4.1
Give - Zapier: by WordImpress – 1.1.2

### Active MU Plugins ###

Local by Flywheel Relative URL (for Live Links): by Flywheel – 1.0

### Theme ###

Name: Twenty Seventeen
Version: 1.4
Author URL: https://wordpress.org/
Child Theme: No – If you're modifying Give on a parent theme you didn't build personally, then we recommend using a child theme. See: How to Create a Child Theme

```
</details>

### PR DESCRIPTION
**Pull request by Sidsector9**
Closes #2853 and #2869 

## Description
By default, Give's front-end notices are auto-dismissible at 5 seconds mark. 

The following code: 
```php
Give_Notices::print_frontend_notice( __( 'Auto dismissible notice', 'give' ), true );
```
will dismiss a front-end notice in 5 seconds.

To have a manually dismissible front-end notice, the above function will require developers to pass additional parameters to the `array()` argument, for example:
```php
Give_Notices::print_frontend_notice( __( 'This is a dismissible notice', 'give' ), true, 'warning', array( 'dismiss_type' => 'manual' ) );
```

## How Has This Been Tested?
Manually tested.

## Screenshots (jpeg or gifs if applicable):
image

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style.
- [ ] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by DevinWalker** [comment]
@Sidsector9 thanks this is looking nice but what about non-dismissible notices like we have by default now? We should preserve that functionality.

**Comment by Sidsector9** [comment]
@DevinWalker I've updated it

### 1. Non Dismissible (default)
```php
Give_Notices::print_frontend_notice( __( 'This is a non-dismissible notice', 'give' ), true );
```

### 2. Dismissible (default: auto)
```php
Give_Notices::print_frontend_notice( __( 'Sample notice', 'give' ), true, 'warning', array( 'dismissible' => true ) );
```
The above will make the notice auto-dismissible

### 3. Dismissible (manual)
```php
Give_Notices::print_frontend_notice( __( 'Sample notice', 'give' ), true, 'warning', array( 'dismiss_type' => 'manual' ) );
```

File: includes/class-notices.php [hunk 1]
```diff
@@ -619,18 +620,41 @@ public static function print_frontend_notice( $message, $echo = true, $notice_ty
...
+					give_clean( $notice_args['dismiss_type'] ),
+					$message
+				);
+				break;
+
+			default:
+				break;
+		}
+
+		wp_enqueue_style( 'dashicons' );
```
Comment (DevinWalker) [inline includes/class-notices.php]: @Sidsector9 rather than using a dashicon can we use an SVG for the close icon?

Comment (Sidsector9) [reply includes/class-notices.php]: @DevinWalker Yes we can, just thought that using dashicons would make the front-end consistent with the admin screen. I'll make the changes.

**Comment by raftaar1191** [comment]
PR description example is outdated please used the below [comment](https://github.com/WordImpress/Give/pull/2871#issuecomment-370664700) example for test

**Pull request merged**